prov/gni/src/gnix_vc.c

```diff
@@ -1104,10 +1122,11 @@ static int __gnix_vc_hndl_conn_req(struct gnix_cm_nic *cm_nic,
 	 * will be filled in.
 	 */
 	GNIX_TRACE(FI_LOG_EP_CTRL, "\n");
+	fastlock_acquire(&ep->vc_lock);
```

Comment:
Why does this lock need to be here now?  Can you put in comments to document for the future?